runtime/lua/vim/lsp/completion.lua

```diff
@@ -0,0 +1,269 @@
+     * will only insert `sole`. Therefore it is recommended to use `textEdit` instead
+     * since it avoids additional client side interpretation.
+     */
+    insertText?: string;
+
+    /**
+     * The format of the insert text. The format applies to both the `insertText` property
+     * and the `newText` property of a provided `textEdit`. If omitted defaults to
+     * `InsertTextFormat.PlainText`.
+     */
+    insertTextFormat?: InsertTextFormat;
+
+    /**
+     * An edit which is applied to a document when selecting this completion. When an edit is provided the value of
+     * `insertText` is ignored.
+     *
+     * *Note:* The range of the edit must be a single line range and it must contain the position at which completion
+     * has been requested.
+     */
+    textEdit?: TextEdit;
+
+    TODO: Handle additionalTextEdits
+    /**
+     * An optional array of additional text edits that are applied when
+     * selecting this completion. Edits must not overlap (including the same insert position)
+     * with the main edit nor with themselves.
+     *
+     * Additional text edits should be used to change text unrelated to the current cursor position
+     * (for example adding an import statement at the top of the file if the completion item will
+     * insert an unqualified type).
+     */
+    additionalTextEdits?: TextEdit[];
+
+    TODO: Handle commitCharacters
+    /**
+     * An optional set of characters that when pressed while this completion is active will accept it first and
+     * then type that character. *Note* that all commit characters should have `length=1` and that superfluous
+     * characters will be ignored.
+     */
+    commitCharacters?: string[];
+
+    TODO: Handle command
+    /**
+     * An optional command that is executed *after* inserting this completion. *Note* that
+     * additional modifications to the current document should be described with the
+     * additionalTextEdits-property.
+     */
+    command?: Command;
+
+    TODO: Handle data
+    /**
+     * A data entry field that is preserved on a completion item between
+     * a completion and a completion resolve request.
+     */
+    data?: any
+}
+--]==]
+
+local completion = {}
+
+--- Extract the completion items from a `textDocument/completion` request
+--@param result (table) The result of a `textDocument/completion` request
+--@returns (table) CompletionItem[]
+function completion.extract_completion_items(result)
+  if type(result) == 'table' and result.items then
+    -- `CompletionList`
+    return result.items
+  elseif result ~= nil then
+    -- `CompletionItem[]`
+    return result
+  else
+    -- result is `null`
+    return {}
+  end
+end
+
+--- Sorts by CompletionItem.sortText.
+--@note: This sorts in-place.
+function completion.sort_completion_items(items)
+  -- TODO: perhaps we don't handle `falsy` values well here.
+
+  table.sort(items, function(a, b)
+    return (a.sortText or a.label) < (b.sortText or b.label)
```

Comment:
If you're worried about falsy things then throw another `or ""` on there.

Reply:
Ah, this probably wasn't clear.

I wasn't sure if `sortText = ""` could be sent by the LSP and be considered `falsy` (and then, obviously not considered `falsy` in Lua. Because then we'd use that to compare the other items.